core-primitives/stf-executor/src/executor.rs

```diff
@@ -187,21 +191,32 @@ where
 		header: &ParentchainHeader,
 		parentchain_id: &ParentchainId,
 	) -> Result<()> {
-		debug!("Update STF storage upon block import!");
-		let storage_hashes = Stf::storage_hashes_to_update_on_block(parentchain_id);
-
-		// global requests they are the same for every shard
-		let state_diff_update = self
-			.ocall_api
-			.get_multiple_storages_verified(storage_hashes, header, parentchain_id)
-			.map(into_map)?;
-
-		// Update parentchain block on all states.
-		// TODO: Investigate if this is still necessary. We load and clone the entire state here,
-		// which scales badly for increasing state size.
 		let shards = self.state_handler.list_shards()?;
-		for shard_id in shards {
+		if let Some(shard_id) = shards.get(0) {
+			debug!("Update STF storage upon block import!");
 			let (state_lock, mut state) = self.state_handler.load_for_mutation(&shard_id)?;
+			let storage_hashes =
+				Stf::storage_hashes_to_update_on_block(&mut state, parentchain_id, &shard_id);
+			trace!(
+				"parentchain storage_hash to mirror: 0x{} at header 0x{}",
+				hex::encode(storage_hashes[0].clone()),
+				hex::encode(header.hash().encode())
+			);
```

Comment:
Same here, we should be able to format this in hex without manually hex encoding.